services/management/grpc/actions_server.go

```diff
@@ -266,3 +269,33 @@ func (s *actionsServer) CancelAction(ctx context.Context, req *managementpb.Canc
 
 	return &managementpb.CancelActionResponse{}, nil
 }
+
+// StartPTSummaryAction starts pt-summary action.
+//nolint:lll
+func (s *actionsServer) StartPTSummaryAction(ctx context.Context, req *managementpb.StartPTSummaryActionRequest) (*managementpb.StartPTSummaryActionResponse, error) {
+	agent, err := models.FindAgentByID(s.db.Querier, req.PmmAgentId)
```

Comment:
find pmm-agents by field RunsOnNodeID which should be equal to req.NodeID. For it use `models.FindPMMAgentsRunningOnNode`.
Then check that passed pmm-agent id in the list by using `models.FindPmmAgentIDToRunAction`.